packages/suite/src/components/suite/DeviceDisplay/parseTextToPagesAndLines.ts

```diff
@@ -0,0 +1,84 @@
+import { DeviceModelInternal } from '@trezor/connect';
+import {
+    MAX_ROWS_PER_PAGE,
+    MAX_CHARACTERS_ON_ROW,
+    CHARACTER_OFFSET_FOR_CONTINUES_ARROW,
+    CHARACTER_OFFSET_FOR_NEXT_ARROW,
+} from '../../../constants/suite/device';
+
+export type ParseTextToLinesParams = {
+    deviceModel: DeviceModelInternal;
+    text: string;
+};
+
+export type ResultRow = { text: string };
+export type ResultPage = { rows: ResultRow[] };
+
+export type ParseTextToLinesResult = {
+    pages: ResultPage[];
+    hasNextIcon: boolean;
+};
+
+export const parseTextToPagesAndLines = ({
+    deviceModel,
+    text,
+}: ParseTextToLinesParams): ParseTextToLinesResult => {
+    const rowsPerPage = MAX_ROWS_PER_PAGE[deviceModel];
+    const charsPerRow = MAX_CHARACTERS_ON_ROW[deviceModel];
+    const offsetForContinuesArrows = CHARACTER_OFFSET_FOR_CONTINUES_ARROW[deviceModel];
+    const offsetForNextArrows = CHARACTER_OFFSET_FOR_NEXT_ARROW[deviceModel];
+
+    const charsOnLastPage = rowsPerPage * charsPerRow - offsetForNextArrows;
+
+    let pageIndex = 0;
+    let remaining = text;
+    let pages: ResultPage[] = [];
+
+    while (true) {
+        const isFirstPage = pageIndex === 0;
+        const isLastPage = remaining.length <= charsOnLastPage;
+
+        let rowIndex = 0;
+        const rows: ResultRow[] = [];
+
+        while (rowIndex < rowsPerPage) {
+            const isFirstRow = rowIndex === 0;
+            const isLastRow = rowIndex === rowsPerPage - 1;
+
+            const isShorterFirstRow = isFirstRow && !isFirstPage;
+            const isShorterLastRow = isLastRow && !isLastPage;
+
+            const charsShorter =
+                // eslint-disable-next-line no-nested-ternary
+                isShorterFirstRow
+                    ? offsetForNextArrows
+                    : isShorterLastRow
+                      ? offsetForContinuesArrows
+                      : 0;
+
+            const chars = charsPerRow - charsShorter;
+
+            rows.push({ text: remaining.slice(0, chars) });
+            remaining = remaining.slice(chars);
+
+            if (remaining.length === 0) {
+                break;
+            }
+
+            rowIndex++;
+        }
+
+        pages.push({ rows });
+
+        if (remaining.length === 0) {
+            break;
+        }
+
+        pageIndex++;
```

Comment:
related to the previous comment:
```suggestion
        if (remaining.length) {
            pageIndex++;
        }
```